src/main/java/org/generationcp/middleware/service/api/dataset/DatasetService.java

```diff
@@ -412,4 +412,6 @@ FilteredPhenotypesInstancesCountDTO countFilteredInstancesAndPhenotypes(
 	List<InstanceDetailsDTO> getInstanceDetails(Integer datasetId, Integer studyId);
 
 	void replaceObservationUnitEntry(List<Integer> observationUnitIds, Integer newEntryId);
+
+	Long getDatasetObservationUnitCount(Integer datasetId);
```

Comment:
Better to name as `countObservationUnits` (we can omit dataset given the datasetId parameter)